conda_build/jinja_context.py

```diff
@@ -494,34 +494,38 @@ def native_compiler(language, config):
     return compiler
 
 
-def compiler(language, config, permit_undefined_jinja=False):
-    """Support configuration of compilers.  This is somewhat platform specific.
+def _target(component, language, config, permit_undefined_jinja=False):
+    """Support configuration of compilers/stdlib.  This is somewhat platform specific.
 
-    Native compilers never list their host - it is always implied.  Generally, they are
+    Native compilers/stdlib never list their host - it is always implied.  Generally, they are
     metapackages, pointing at a package that does specify the host.  These in turn may be
     metapackages, pointing at a package where the host is the same as the target (both being the
     native architecture).
     """
 
-    compiler = native_compiler(language, config)
+    if component == "compiler":
+        package_prefix = native_compiler(language, config)
+    else:
+        package_prefix = language
+
     version = None
     if config.variant:
         target_platform = config.variant.get("target_platform", config.subdir)
-        language_compiler_key = f"{language}_compiler"
-        # fall back to native if language-compiler is not explicitly set in variant
-        compiler = config.variant.get(language_compiler_key, compiler)
-        version = config.variant.get(language_compiler_key + "_version")
+        language_key = f"{language}_{component}"
+        # fall back to native if language-key is not explicitly set in variant
+        package_prefix = config.variant.get(language_key, package_prefix)
```

Comment:
Not sure if the fallback makes sense for `stdlib`? If I read the code correctly, the fallback key we'd be looking for would just be `c`?

IOW, if I'm not overlooking something here, I think we could move move the `if "compiler"` branch in here rather than setting a dummy `package_prefix` for `stdlib`.

Reply:
Yes, it would be `c_linux-64`. We can also use `{{ compiler('clang') }}` which expands to `clang_linux-64` with this fallback. Don't see a reason we shouldn't allow it for stdlib.